lib/datadog/core/configuration/options.rb

```diff
@@ -68,18 +68,15 @@ def options
           end
 
           def set_option(name, value, precedence: Configuration::Option::Precedence::PROGRAMMATIC)
-            add_option(name) unless options.key?(name)
-            options[name].set(value, precedence: precedence)
+            resolve_option(name).set(value, precedence: precedence)
```

Comment:
This is a small refactor to this file, because `add_option(name) unless options.key?(name)` was too common and not trivial to understand.